BnbnavNetClient/Views/MapView.axaml.cs

```diff
@@ -52,20 +59,64 @@ protected override void OnInitialized()
 
             // We need to pan _more_ when scale is smaller:
             MapViewModel.Pan += (_pointerPrevPosition - pointerPos) / MapViewModel.Scale;
+            
+            _pointerVelocities.Add(new Point(
+                _pointerPrevPosition.X - pointerPos.X,
+                _pointerPrevPosition.Y - pointerPos.Y
+            ));
+
+            if (_pointerVelocities.Count > 5)
+                _pointerVelocities.RemoveAt(0);
+
+            var xSum = 0.0;
+            var ySum = 0.0;
+
+            foreach (var i in _pointerVelocities) {
+                xSum += i.X;
+                ySum += i.Y;
+            }
+
+            _viewVelocity = new(
+                xSum / _pointerVelocities.Count,
+                ySum / _pointerVelocities.Count
+            );
+
+            /*
+             * The actual view velocity should be the average of the last 5
+             * computed velocities, due to how low-quality mouses work (low-quality
+             * mouses have a tendency to move in angles snapped to 45 degrees).
+             */
+
             _pointerPrevPosition = pointerPos;
         };
 
         PointerReleased += (_, __) =>
         {
             _pointerPressing = false;
+            _pointerVelocities.Clear(); // Make sure we're not using velocities from previous pan.
         };
 
         PointerWheelChanged += (_, eventArgs) =>
         {
             var deltaScale = eventArgs.Delta.Y * MapViewModel.Scale / 10.0;
             Zoom(deltaScale, (eventArgs.GetPosition(this)));
+
+            _viewVelocity = new(); // Reset velocities
+            _pointerVelocities.Clear();
         };
 
+        // This is the physics part of inertial panning.
+        Clock = new Clock();
+        Clock.Subscribe(
+            ts => {
+                if (_pointerPressing)
+                    return;
+            
+                MapViewModel.Pan += _viewVelocity / MapViewModel.Scale;
+                _viewVelocity /= 1.075; // 1.075 is the friction.
```

Comment:
Can we make sure that at some threshold this decays to 0 and the timer stops? Otherwise we'll be asking Avalonia to draw the frame unnecessarily often

Reply:
How do I stop the timer? And how do I run it again?

Reply:
My main concern is that the pan is constantly being updated - that is what forces a redraw of the screen. If you can avoid setting the pan below a certain threshold that would be acceptable for me.

As for stopping the clock, @Vrabbers do you have any tips?

Reply:
Actually, yes. I don't need to stop the timer, as long as I avoid the pan.